docs/quickstart.fsx

```diff
@@ -0,0 +1,229 @@
+#endif // IPYNB
+(*** condition: ipynb ***)
+#if IPYNB
+// Import DiffSharp package
+#r "nuget: DiffSharp-lite,{{fsdocs-package-version}}"
+#r "nuget: SixLabors.ImageSharp,1.0.1"
+
+// Set dotnet interactive formatter to plaintext
+Formatter.SetPreferredMimeTypeFor(typeof<obj>, "text/plain")
+Formatter.Register(fun (x:obj) (writer: TextWriter) -> fprintfn writer "%120A" x )
+#endif // IPYNB
+
+(**
+[![Binder](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/DiffSharp/diffsharp.github.io/blob/master/{{fsdocs-source-basename}}.ipynb)&emsp;
+[![Binder](img/badge-binder.svg)](https://mybinder.org/v2/gh/diffsharp/diffsharp.github.io/master?filepath={{fsdocs-source-basename}}.ipynb)&emsp;
+[![Script](img/badge-script.svg)]({{fsdocs-source-basename}}.fsx)&emsp;
+[![Script](img/badge-notebook.svg)]({{fsdocs-source-basename}}.ipynb)
+
+# Quickstart
+
+Here we cover some key tasks involved in a typical machine learning pipeline and how these can be implemented with DiffSharp. Note that a significant part of DiffSharp's design has been influenced by [PyTorch](https://pytorch.org/) and you would feel mostly at home if you have familiarity with PyTorch.
+
+## Datasets and Data Loaders
+
+DiffSharp provides the `cref:T:DiffSharp.Data.Dataset` type that represents a data source and the `cref:T:DiffSharp.Data.DataLoader` type that handles the loading of data from datasets and iterating over [minibatches](https://en.wikipedia.org/wiki/Stochastic_gradient_descent#Iterative_method) of data.
+
+See the [DiffSharp.Data](/reference/diffsharp-data.html) namespace for the full API reference.
+
+### Datasets
+
+DiffSharp has ready-to-use types that cover main datasets typically used in machine learning, such as `cref:T:DiffSharp.Data.MNIST`, `cref:T:DiffSharp.Data.CIFAR10`, `cref:T:DiffSharp.Data.CIFAR100`, and also more generic dataset types such as `cref:T:DiffSharp.Data.TensorDataset` or `cref:T:DiffSharp.Data.ImageDataset`.
+
+The following loads the [MNIST](http://yann.lecun.com/exdb/mnist/) dataset and shows one image entry and the corresponding label.
+*)
+
+open DiffSharp
+open DiffSharp.Data
+
+// First ten images in MNIST training set
+let dataset = MNIST("../data", train=true, transform=id, n=10)
+
+// Inspect a single image and label
+let data, label = dataset.[7]
+
+// Save image to file
+data.saveImage("test.png")
+
+(** *)
+
+(*** hide ***)
+pngToHtml "test.png"
+(*** include-it-raw ***)
+
+// Inspect data as ASCII and show label
+printfn "Data: %A\nLabel: %A" (data.toImageString()) label
+(*** include-output ***)
+
+(**
+ 
+### Data Loaders
+
+A data loader handles tasks such as constructing minibatches from an underlying dataset on-the-fly, shuffling the data, and moving the data tensors between devices. In the example below we show a single batch of six MNIST images and their corresponding classification labels.
```

Comment:
More information on devices and data loaders would be useful, e.g. guidance on how many tensor copies to expect when getting data to GPU etc.

Reply:
The idea is that the quickstart page is just about the basics and getting up and running. I think we can have the full "using devices" guidance in another page and link to it from here. Alternatively we can also cover GPU usage in quickstart.